frontend/game-rating-platform/src/compoments/GameTable.tsx

```diff
@@ -23,21 +16,42 @@ export function GameTable({
   // cache original game data for filter operations
   const [origGameData, setOrigGameData] = useState<GameProps[]>([]);
   const [gameData, setGameData] = useState<GameProps[]>([]);
+  const [activeModal, setActiveModal] = useState<boolean>(false);
+  const [clickedGame, setClickedGame] = useState<GameProps>({
+    id: 0,
+    title: "",
+    releaseDate: "",
+    developers: [],
+    description: "",
+    trailer: "",
+    avgRating: 0,
+    platform: "",
+  });
+
+  function openModal(e: SyntheticEvent<HTMLTableRowElement, Event>) {
+    setActiveModal(true);
+    let filteredData = gameData.filter(
+      (prop) => prop.title === e.currentTarget.id
+    );
+    console.log(filteredData);
+    if (filteredData.length !== 0) {
+      setClickedGame(filteredData[0]);
+    }
+  }
+
+  function closeModal() {
+    setActiveModal(false);
+  }
 
   useEffect(() => {
     const fetchGames = async (requestUrl: string) => {
-      const response = await fetch(requestUrl, {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-        },
+      let data: GameProps[] = await fetchGameData(requestUrl);
+      data = data.map((props) => {
+        props.avgRating = Number.isNaN(Number(props.avgRating))
```

Comment:
Das ist eine Logik, die ich wahrscheinlich eher im Backend erwartet hätte, z.B. bei der Umwandlung vom `Game` in ein `GameDTO`. Die Funktion `fetchGames()` würde damit zu einem Einzeiler werden, weil du dir diese Transformation sparen kannst, nämlich

```typescript
const fetchGames = async (requestUrl: string) => {
  setOrigGameData(await fetchGameData(requestUrl));
};
```

Ist aber mit Sicherheit diskutabel, da ja auch sowas wie "N/A" statt `0` als Darstellung im Frontend denkbar wäre, was nicht möglich wäre, wenn du die Daten aus dem Backend bereits als `0` sendest.